### ISSUE TITLE
Starting a process instance by message event doesn't work if the message id and message name differ

### ISSUE THREAD

**Issue by Adam-Higginson** [Activiti/Activiti#2271]
If you attempt to start a process instance by message by calling `startProcessInstanceByMessage` as per the docs [found here](https://www.activiti.org/userguide/#bpmnMessageStartEvent) it is expected you will pass in the message name. However, in the bpmn file when linking the start event to the message you link by message id. This then causes a `ActivitiException` to be thrown of the form: `No message start event found for process definition <something> and message name <somethingElse>`

As an example of this:
```
<message id="myMessageId" name="myMessageName" />

<process id="MyProcess" name="My Process">

<startEvent id="Start" name="Start">
   <messageEventDefinition messageRef="myMessageId" />
</startEvent>
 
<!-- Process continues... -->
```

And then in the java code:
```
processEngine.getRuntimeService().startProcessInstanceByMessage("myMessageName")
```

This eventually calls through to `ProcessInstanceHelper` which tried to find the corresponding message by doing something like:

```
          if (messageEventDefinition.getMessageRef().equals(messageName)) {
            initialFlowElement = flowElement;
            break;
          }
```

However, this uses the message ref of the start event which happens to be the message id, not the message name and thus an exception is thrown.

**Comment by salaboy** [comment]
@Adam-Higginson are you using Activiti 6.x?

**Comment by Adam-Higginson** [comment]
Yep, I noticed this when trying to update an Activiti 5 setup to Activiti 6

**Comment by Abdullah8006** [comment]
**flowable-spring-boot-starter** 6.4.0 has the same issue.

**Comment by yangcnb** [comment]
Is this the author's original intention？I also encountered this problem

**Comment by ming820214** [comment]
//消息传参传的是消息值，这里的判断也应该以消息值为准，这样的判断才有意义，消息值相同的节点就可以进行相应的后续操作了。 edit by peter zhang
                    String messageRef = messageEventDefinition.getMessageRef();
                    if(bpmnModel.containsMessageId(messageRef)) {
                        Message message = bpmnModel.getMessage(messageRef);
                        if (message.getName().equals(messageName)) {
                            initialFlowElement = flowElement;
                            break;
                        }
                    }

> If you attempt to start a process instance by message by calling `startProcessInstanceByMessage` as per the docs [found here](https://www.activiti.org/userguide/#bpmnMessageStartEvent) it is expected you will pass in the message name. However, in the bpmn file when linking the start event to the message you link by message id. This then causes a `ActivitiException` to be thrown of the form: `No message start event found for process definition <something> and message name <somethingElse>`
> 
> As an example of this:
> 
> ```
> <message id="myMessageId" name="myMessageName" />
> 
> <process id="MyProcess" name="My Process">
> 
> <startEvent id="Start" name="Start">
>    <messageEventDefinition messageRef="myMessageId" />
> </startEvent>
>  
> 
> ```
> 
> And then in the java code:
> 
> ```
> processEngine.getRuntimeService().startProcessInstanceByMessage("myMessageName")
> ```
> 
> This eventually calls through to `ProcessInstanceHelper` which tried to find the corresponding message by doing something like:
> 
> ```
>           if (messageEventDefinition.getMessageRef().equals(messageName)) {
>             initialFlowElement = flowElement;
>             break;
>           }
> ```
> 
> However, this uses the message ref of the start event which happens to be the message id, not the message name and thus an exception is thrown.

将源码进行如下修改就可以了
org.activiti.engine.impl.util.ProcessInstanceHelper

 //消息传参传的是消息值，这里的判断也应该以消息值为准，这样的判断才有意义，消息值相同的节点就可以进行相应的后续操作了。 edit by peter zhang
                    String messageRef = messageEventDefinition.getMessageRef();
                    if(bpmnModel.containsMessageId(messageRef)) {
                        Message message = bpmnModel.getMessage(messageRef);
                        if (message.getName().equals(messageName)) {
                            initialFlowElement = flowElement;
                            break;
                        }
                    }
